R/target_market_share.R

```diff
@@ -127,102 +127,62 @@ target_market_share <- function(data,
   )
 
   check_valid_columns(data, valid_columns)
-  data <- aggregate_by_loan_id(data)
 
   crucial_scenario <- c("scenario", "tmsr", "smsp")
   check_crucial_names(scenario, crucial_scenario)
   check_crucial_names(ald, "is_ultimate_owner")
   walk_(crucial_scenario, ~ check_no_value_is_missing(scenario, .x))
 
   green_or_brown <- r2dii.data::green_or_brown
+  tmsr_or_smsp <- tmsr_or_smsp()
```

Comment:
This dataset is defined by a function, and called directly later in the script. I thought it would make sense to specify explicitly where the dataset is coming from when we import the other necessary datasets.

Reply:
Sure that makes sense. My hope was to elucidate that data was necessary that doesn't necessarily come from an argument, and to clearly define it, but I see the point. I'll move them closer to where they are used.